src/components/LyricView.tsx

```diff
@@ -59,29 +59,51 @@ const LyricView: React.FC<LyricViewProps> = (props: LyricViewProps) => {
     let key = 0;
 
     const songLyrics = song.lyrics;
-    const verseKeys = Object.keys(songLyrics);
-    const lyricBlocks: string[][] = Object.values(songLyrics);
+    const presentationOrder = song.presentation?.split(" ");
 
-    for (let i = 0; i < verseKeys.length; ++i) {
-      const verseKey = verseKeys[i];
-      const verseLyrics = lyricBlocks[i];
+    // if the song has a presentation order defined, use it;
+    // otherwise, present in written order
+    if (typeof presentationOrder !== "undefined") {
+      presentationOrder.forEach((verseNumber) => {
+        const verseName: string = getVerseText(verseNumber);
+        const verseLyrics: string[] = songLyrics.get(verseNumber) || [""];
 
-      lyrics.push(
-        <Fragment key={key++}>
-          <div className="ion-margin-vertical"></div>
-        </Fragment>
-      );
-      lyrics.push(
-        <h5 key={key++} className="ion-margin-top">
-          {getVerseText(verseKey)}
-        </h5>
-      );
-      verseLyrics.forEach((line: string) => {
-        lyrics.push(
-          <IonText key={key++} className="lyricVerse" color="dark">
-            <p>{line}</p>
-          </IonText>
-        );
+        lyrics.push(buildLyricBlock(verseName, verseLyrics, key++));
+      });
+    } else {
```

Comment:
I actually couldn't find any songs without a presentation populated, which makes this difficult to write a test for. Did you at least test that this case works locally? If so, I think that's good enough :)